st2reactor/st2reactor/container/process_container.py

```diff
@@ -290,3 +311,10 @@ def _dispatch_trigger_for_sensor_exit(self, sensor, exit_code):
             'exit_code': exit_code
         }
         self._dispatcher.dispatch(trigger, payload=payload)
+
+    def _delete_sensors(self, sensor_id):
+        with self.lock:
```

Comment:
Why do we need a lock here?

Eventlet's green threads are co-operative and we don't use traditional threads here so lock shouldn't be needed here (especially using a lock from stdlib threading module might result in some weird cases).